roofit/histfactory/src/HistFactoryNavigation.cxx

```diff
@@ -839,16 +817,14 @@ namespace RooStats {
    // Based on the mode, we assume that node is
    // the "unconstrained" pdf node for that channel
    RooArgSet* components = pdf->getComponents();
-   TIter argItr = components->createIterator();
-   RooAbsArg* arg = NULL;
-   while( (arg=(RooAbsArg*)argItr.Next()) ) {
-     std::string ClassName = arg->ClassName();
-     if( ClassName == "RooRealSumPdf" ) {
-       fChannelSumNodeMap[ChannelName] = (RooRealSumPdf*) arg;
-       break;
-     }
-   }
+    for (auto const *arg : static_range_cast<RooAbsArg *>(*components)) {
```

Comment:
```suggestion
    for (auto const *arg : *pdf->getComponents()) {
```
Unnecessary cast, and I guess you can collapse this with the previous line that can be removed?